### PR TITLE
Add bokeh

### PR DESCRIPTION
**Pull request by jakirkham**
Closes https://github.com/conda-forge/staged-recipes/pull/1055

Adds a recipe to build `bokeh`. Generated via `conda skeleton pypi` and tweaked. Starting with 0.11.1, but will bump to 0.12.0 in the feedstock. Feedback welcome. Also, more maintainers welcome.

cc @bryevdv @mattpap @damianavila @birdsarah @fpliger @canavandl

### CODE REVIEW

**Comment by Korijn** [comment]
I'm guessing @almarklein wants in on this one, maybe?

**Comment by jakirkham** [comment]
Weird. It uses `setuptools`, but not for `setup`? How did I miss this?

**Comment by cel4** [comment]
Seems that jinja2 is missing the setuptools dependency at runtime.

http://jinja.pocoo.org/docs/dev/intro/#prerequisites

> If you wish to use the PackageLoader class, you will also need setuptools or distribute installed at runtime.

**Comment by jakirkham** [comment]
Please add an issue and/or PR to that feedstock, @cel4.

**Comment by jakirkham** [comment]
Went ahead and raised the `jinja` issue ( https://github.com/conda-forge/jinja2-feedstock/issues/3 ) and proposed a fix with PR ( https://github.com/conda-forge/jinja2-feedstock/pull/4 ).

File: recipes/bokeh/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    - bokeh.util
+
+about:
+  home: http://github.com/bokeh/bokeh
+  license: BSD 3-Clause
+  summary: Statistical and novel interactive HTML plots for Python
+
+extra:
+  recipe-maintainers:
+    - almarklein
```
Comment (jakirkham) [inline recipes/bokeh/meta.yaml]: Added you here @almarklein. 😄

Comment (bryevdv) [inline recipes/bokeh/meta.yaml]: Just curious... what does the Node install and then `npm install` in the `bokehjs` dir? BokehJS has to be built from coffeescript sources.

Comment (bryevdv) [inline recipes/bokeh/meta.yaml]: which also means you probably want `python setup.py install --build-js`

File: recipes/bokeh/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+{% set version = "0.11.1" %}
+{% set checksum = "34a73d8f68e7d6faaecc9f785cac8e9f76b88a7e7aacf66307076b03b4d3c472" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
```
Comment (jakirkham) [inline recipes/bokeh/meta.yaml]: Was getting the source from PyPI @bryevdv in the assumption that the Coffescript was already compiled and included for a first pass. However, it seems the JS and Coffescript for that matter are not present in the PyPI. Might be worth investigating.

Comment (jakirkham) [inline recipes/bokeh/meta.yaml]: > Maybe you are looking in the wrong place?

:facepalm: 😳 Yep, I'm as blind as a bat. Thanks for pointing it out to me.

File: recipes/bokeh/meta.yaml (hunk 1 above)
Comment (almarklein) [inline recipes/bokeh/meta.yaml]: > Added you here @almarklein. :smile:

I suspect @bryevdv also wants to be on this list ;)

Comment (bryevdv) [inline recipes/bokeh/meta.yaml]: I have more than enough on my plate I am happy to spectate here

Comment (damianavila) [inline recipes/bokeh/meta.yaml]: > Just curious... what does the Node install and then npm install in the bokehjs dir? BokehJS has to be built from coffeescript sources.

If you are getting the source from pypi, there is no need for node steps... All the necessary JS and CSS should be there in the tar.gz package.

In this case they are building from our sources, not from scratch... which I think it is a reasonable approach...

File: recipes/bokeh/meta.yaml (hunk 2 above)
Comment (damianavila) [inline recipes/bokeh/meta.yaml]: Jaja... Poor @bryevdv's 💔

File: recipes/bokeh/meta.yaml (hunk 1 above)
Comment (jakirkham) [inline recipes/bokeh/meta.yaml]: > I have more than enough on my plate I am happy to spectate here

Honestly, when asking devs to sign up as maintainers, this is sort of what I'm hoping for basically. The exact points that you have raised here about how to properly build the package are what I want to know. So, if I do something silly, someone who knows the codebase better can point it out to me. If either of you would be willing to that @bryevdv @damianavila, then I would be more than happy to add you here.

Comment (jakirkham) [inline recipes/bokeh/meta.yaml]: > In this case they are building from our sources, not from scratch... which I think it is a reasonable approach...

Yeah, I was thinking about building from scratch, but wasn't ready to dive through the dependency tree on a first pass. Though, in the long run, I'd like to explore it in the feedstock.

**Comment by ahmadia** [comment]
Shall we close this in favor of https://github.com/conda-forge/staged-recipes/pull/1055?

**Comment by jakirkham** [comment]
Closing in favor of PR ( https://github.com/conda-forge/staged-recipes/pull/1055 ). Thanks @ahmadia for taking over.

**Comment by jakirkham** [comment]
If someone could review and merge this (assuming it is passing), that would be much appreciated.

**Comment by almarklein** [comment]
Maybe I missed it, but what happened to #1055? It seems about ready too?

**Comment by jakirkham** [comment]
Yeah, but there were a couple of things to do and the contributor doesn't seem to be working on it actively any more.

**Comment by scopatz** [comment]
Mind merging from master @jakirkham?

**Comment by jakirkham** [comment]
Sure, have rebased on current `master`.

**Comment by lexual** [comment]
Can I do anything to help push this out the door ;)

**Comment by jakirkham** [comment]
Alright, this has sat far too long. Going to go ahead and merge. Feel free to submit follow-up PRs to the feedstock for any other changes.

**Pull request merged**